### PR TITLE
fix(gatsby-cli): don't crash when creating project from local starter

### LINKED ISSUE #16065: Issue when attempting to create a new project from a local starter
## Description

Gatsby CLI fails when creating a project from a local starter.

I've encountered this issue when trying to create a starter that cannot be hosted on github.com.

Since `gatsby` don't support non-public repository hosts (yet, I hope to create a PR that will help on this later), creating a project from a local starter is currently the only way, but I when I tested I found this issue.

### Steps to reproduce

1. clone the default starter at https://github.com/gatsbyjs/gatsby-starter-default.git

2. attempt to create a project based on the local starter: `gatsby new my-project /path/to/local/gatsby-starter-default`

### Expected result

A new project is created using the local starter as a template.

### Actual result

No project was created and the following error is thrown:

```
  TypeError: Cannot read property 'shortcut' of undefined

  - init-starter.js:357
    [lib]/[gatsby]/[gatsby-cli]/lib/init-starter.js:357:31
```

### Environment

```
  System:
    OS: Linux 4.4 Ubuntu 16.04.3 LTS (Xenial Xerus)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 4.3.48 - /bin/bash
  Binaries:
    Node: 8.11.1 - ~/.nvm/versions/node/v8.11.1/bin/node
    Yarn: 1.6.0 - ~/.nvm/versions/node/v8.11.1/bin/yarn
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.1/bin/npm
```

### PR DESCRIPTION
**Pull request by marcos-abreu**
## Description

Fix issue when attempting to create a new project using a local starter.

## Related Issues

Fixes #16065

### CODE REVIEW

File: docs/tutorial/part-one/index.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ gatsby new [SITE_DIRECTORY_NAME] [URL_OF_STARTER_GITHUB_REPO]
 
 If you omit a URL from the end, Gatsby will automatically generate a site for you based on the [**default starter**](https://github.com/gatsbyjs/gatsby-starter-default). For this section of the tutorial, stick with the “Hello World” site you already created in tutorial part zero.
 
+> 💡 You can also create a new site from a local starter by running: `gatsby new [SITE_DIRECTORY_NAME] [PATH/TO/LOCAL/STARTER]`
+
```
Comment (pieh) [inline docs/tutorial/part-one/index.md]: Can we make change like this in separate PR and have only actual fix in this one?

We try not to overload tutorial with every possible way you can create new gatsby project, so I don't think this is best place for it. I would put this in https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/gatsby-cli.md

Comment (marcos-abreu) [reply docs/tutorial/part-one/index.md]: I will do it. Thanks for the quick reply

Comment (marcos-abreu) [reply docs/tutorial/part-one/index.md]: I've removed the commit with the docs, leaving only the code fix on this PR as suggested.

I've also looked into your suggestion of opening another PR with the doc change on the `gatsby-cli.md` file instead of the tutorial. But when reading the content of the `gatsby-cli.md` this is what is currently there for the `new` command:

```
### `new`

`gatsby new gatsby-site`

See the [Gatsby starters docs](/docs/starters/)
for a comprehensive list of starters to get started with Gatsby.
```

So I will be submiting another PR, but to the https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/starters.md instead.

**Comment by marcos-abreu** [comment]
@pieh - it looks like the pipeline was canceled. Is there anything I can do on my end to re-run the pipeline?

**Comment by pieh** [comment]
> @pieh - it looks like the pipeline was canceled. Is there anything I can do on my end to re-run the pipeline?

If you are not able to click "rerun workflow" in circleCI dashboard, you could push some commits to trigger tests. But I just restarted those, so hopefully we should be sorted here.

**Comment by wardpeet** [comment]
Fixed in `gatsby-cli@2.7.21`
